src/server/middleware/repoproxy/proxy.go

```diff
@@ -113,19 +113,27 @@ func handleManifest(w http.ResponseWriter, r *http.Request, next http.Handler) e
 	log.Debugf("the tag is %v, digest is %v", art.Tag, art.Digest)
 	man, err := proxyCtl.ProxyManifest(ctx, p, art)
 	if err != nil {
-		return err
+		fallbackToLocal(w, r, next, err)
```

Comment:
Should we check whether the error is not found?

Reply:
OK, return the NotFoundError